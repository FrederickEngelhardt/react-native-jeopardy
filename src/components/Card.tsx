import React from "react";
import { StyleSheet, Text, View } from "react-native";
import { graphql, QueryRenderer } from "react-relay";
import environment from "../environment";

export default function Card() {
  return (
    <View style={styles.container}>
      <QueryRenderer
        environment={environment}
        query={graphql`
          query CardQuery {
            allFilms {
              edges {
                node {
                  id
                  title
                }
              }
            }
          }
        `}
        render={({ error, props }) => {
          if (error) {
            console.warn(error);
            return <Text>Failed</Text>;
          }
          if (!props) {
            return <Text>Loading...</Text>;
          }
          console.log(props);
          return (
            <>
            {props.allFilms.edges.map(({ node }) => {
              console.log(node.title)
              return <Text key={node.id}>{node.title}</Text>
            })}
            </>
            )
        }}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center"
  }
});
