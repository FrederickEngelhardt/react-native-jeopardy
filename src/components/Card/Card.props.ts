const cardProps = {
	points: 500,
	title: 'Basic Title',
	answers: ["Your answer"],
	questionHints: [
		{ title: 'Planets', value: 'Naboo'}
	]
}

export default cardProps
