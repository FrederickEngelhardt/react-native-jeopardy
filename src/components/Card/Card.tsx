import React from "react";
import styled from "styled-components";

const CardContainer = styled.View`
  padding: 24px;
  background-color: ${props => (props.theme.darkMode ? "#6B6B6B" : "#d1d1d1")};
  // width: 80%;
  border-radius: 5px;
`;

const TouchableContainer = styled.TouchableOpacity`
  width: 80%;
  align-self: center;
`;

const TitleText = styled.Text`
  font-size: 38px;
  text-align: center;
  color: ${props => (!props.theme.darkMode ? "#000" : "#fff")};
`;

const QuestionText = styled(TitleText)`
  font-size: 24px;
`;

export const closed = "CLOSED";
export const opened = "OPENED";

interface Props {
  isOpened: boolean
  handleToggle: () => void;
  points: number
  title: string
  questionText: string,
}

const Card = ({ isOpened, title, points, questionText, handleToggle }: Props) => {
  return (
    <TouchableContainer onPress={handleToggle}>
      <CardContainer>
        {!isOpened ? (
          <TitleText>{points}</TitleText>
        ) : (
          <>
            <TitleText>{title}</TitleText>
            <QuestionText>{questionText}</QuestionText>
          </>
        )}
      </CardContainer>
    </TouchableContainer>
  );
};

Card.defaultProps = {
  isOpened: false
}

export default Card;
