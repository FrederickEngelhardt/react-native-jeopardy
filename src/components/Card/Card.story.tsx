import { storiesOf } from "@storybook/react-native";
import * as React from "react";
import { ThemeProvider } from "styled-components";


import Card from "./Card";
import cardProps from "./Card.props";
import {
  boolean,
  number,
  select,
  text,
  withKnobs
} from "@storybook/addon-knobs";

storiesOf("Card", module)
  .addDecorator(withKnobs)
  .add("default", () => (
    <Card
      {...cardProps}
      title={text("Title Text", cardProps.title)}
      isOpened={boolean("Toggle Card Opened", cardProps.isOpened)}
      points={number("Card Points", cardProps.points)}
      handleToggle={() => {}}
    />
  ));
