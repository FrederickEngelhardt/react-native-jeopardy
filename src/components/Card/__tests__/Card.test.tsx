import React from "react";
import { render } from "@testing-library/react-native";

import Card from "../Card";
import cardProps from "../Card.props";

const theme = {
  darkMode: true
};

describe("test", () => {
  it("renders correctly", () => {
    const { container } = render(<Card {...cardProps} theme={theme} />);
    expect(container).toMatchSnapshot()
  });
});
