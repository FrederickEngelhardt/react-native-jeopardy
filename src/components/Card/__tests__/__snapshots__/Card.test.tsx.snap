// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    accessibilityLabel="Jeopardy Topic Card"
    accessible={true}
    activeOpacity={0.5}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "opacity": 1,
          "width": "80%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#d1d1d1",
            "borderRadius": 5,
            "marginBottom": 12,
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 38,
              "fontWeight": "600",
              "textAlign": "center",
            },
          ]
        }
      >
        500
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;
